configure.ac

```diff
@@ -85,6 +85,22 @@ PKG_CHECK_MODULES([DEPS],
                    libxml-2.0
                    protobuf])
 
+AC_PATH_TOOL([GTEST_CONFIG], [gtest-config])
+AC_ARG_VAR([GTEST_CPPFLAGS], [CPPFLAGS for compiling with gtest])
+AC_ARG_VAR([GTEST_LIBS], [LIBS for linking with gtest])
+test -n "$GTEST_CPPFLAGS" || GTEST_CPPFLAGS=$($GTEST_CONFIG --cppflags)
```

Comment:
I forget, does AC_PATH_TOOL abort with an error if the tool is not found? If not we should explicitly die before attempting to run the command. Alternatively if we can fix the default build (without unit tests) to work without gtest/gmock being installed then configure can gracefully skip this when the config tool isn't found.